src/rocblas_wrappers.cc

```diff
@@ -238,6 +238,132 @@ void dotu(
             (rocblas_double_complex*) result));
 }
 
+//------------------------------------------------------------------------------
+// iamax
+//------------------------------------------------------------------------------
+void iamax(
+    int64_t n,
+    float const* dx, int64_t incdx,
+    int64_t* result,
+    blas::Queue& queue )
+{
+    // clear memory if allocated on host
+    if (! is_devptr( result, queue )) {
+        *result = 0;
+    }
```

Comment:
I think result also needs to be cleared if in GPU memory, using `hipMemsetAsync` here, and `cudaMemsetAsync` in the `.cu` file. Repeat 4x for data types.